fuzz/provider.c

```diff
@@ -251,6 +254,22 @@ static uint64_t UITERS = 1;
 static int64_t BLOCKSIZE = 8;
 static uint64_t UBLOCKSIZE = 8;
 
+
+static void free_params(OSSL_PARAM *param)
+{
+    for (; param != NULL && param->key != NULL; param++) {
+        switch (param->data_type) {
+            case OSSL_PARAM_INTEGER:
+            case OSSL_PARAM_UNSIGNED_INTEGER:
+            case OSSL_PARAM_REAL:
+                if (param->data != NULL) {
+                    free(param->data);
+                }
+                break;
+        }
+    }
+}
+
```

Comment:
Isn't this what OSSL_PARAM_free is used for?  Not sure why its duplicated here

Reply:
The implementation only frees when the data_type is `OSSL_PARAM_ALLOCATED_END` which I'm not so sure what it is about. 

The fuzzer dynamically allocates data that requires a certain alignment, which are integers and floats right now. It is not as clean as I wish it would be but for fuzzing I guess it suffices.


https://github.com/openssl/openssl/blob/master/crypto/params_dup.c#L225